### PR TITLE
Bbotk in avglearner 

### LINKED ISSUE #482: Rewrite LearnerAvg with `bbotk`

### PR DESCRIPTION
**Pull request by pfistfl**
closes #482

### CODE REVIEW

**Comment by mb706** [comment]
Please use one of the [github keywords for closing issues through PRs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue that should obviously closed after merging.

**Comment by pfistfl** [comment]
I would merge this once CI passes if there are no objections!

**Comment by mb706** [comment]
please don't merge w/o review

File: R/LearnerAvg.R
Comment (mb706) [inline R/LearnerAvg.R:60]: (1) Since the same checks are done both here and in the RegrAvg, maybe they should be functions in the package namespace.

(2) I don't think `msr(x)` needs to be `check_r6`'d. Also this throws an error if `x` is not in `msr()`, when it only should return TRUE or a character(1). This should probably be in the direction of
```r
if (is.character(x)) {
  if (x %nin% mlr_measures$keys()) "Is a `character` but not a known measure" else TRUE
} else {
  check_r6(x, "MeasureClassif")
}
```

File: R/LearnerAvg.R
Comment (mb706) [inline R/LearnerAvg.R:217]: if `as_measure()` works here use that

File: R/LearnerAvg.R
Comment (mb706) [inline R/LearnerAvg.R:233]: this should be *before* `lgr$set_threshold(self$param_set$values$log_level)` so that things get reset if the optimizer fails (and even in the case where `self$apram_set$values$log_level` is invalid or smth.

File: R/LearnerAvg.R
Comment (mb706) [inline R/LearnerAvg.R:213]: Maybe we want to set the x0 and xtol even if the optimizer was given as an argument (unless these values were set to something else by the user?) Or is that not possible? (i.e. is `x0` a "required" parameter and the user would always have to set these parameters?)

**Review by mb706 — CHANGES_REQUESTED** [review]

File: R/LearnerAvg.R
Comment (pfistfl) [reply R/LearnerAvg.R:217]: There is no `as_measure.character`

File: R/LearnerAvg.R
Comment (pfistfl) [reply R/LearnerAvg.R:213]: `x0` is required for some algorithms e.g. COBYLA but not for all. 
We could do this, but I do not see much added benefit and we might run into strange behavior when e.g. a different algorithm is set.

I just wanted to have this be the default as it IIRC reflects what worked best in Ledell, 2015.

File: R/LearnerAvg.R
Comment (pfistfl) [reply R/LearnerAvg.R:60]: I factored them out.

File: R/LearnerAvg.R
Comment (mb706) [reply R/LearnerAvg.R:195]: huh ok, thought this is what `as_measure` was good for.

**Pull request merged**
